### DISCUSSION TITLE
Do you know a program called ruffle?

### DISCUSSION THREAD

**Discussion by semoki** [airsdk/Adobe-Runtime-Support#3482] [General]
Do you know a program called ruffle? I accidentally discovered ruffle while surfing the web and tested it, and it ran quite well. I saw that Harman was considering web support the other day, so I was wondering what your plans were.

**Comment by ylazy** [comment]
[Ruffle](https://ruffle.rs/) uses WebAssembly so the performance is very good.
[AS3 Progress](https://ruffle.rs/compatibility/avm2)

**Comment by ajwfrost** [comment]
Yes - when we started on all this, Ruffle was reasonable for AS2 but really lacking in AS3 support. It's great to see that they've really improved things - I have to confess I was expecting the efforts to die down a bit after the Flash Player got withdrawn, seeing as their focus was on Flash (web) content and a lot of that got taken down/replaced by HTML/JS/etc.

The web support we're planning for AIR would also use WebAssembly, and would then have a number of the limitations that the Flash Player - and Ruffle - face from being within a browser context. But it would mean that a lot of apps could then be a bit more portable across platforms..

thanks
